requesttojson/requesttojson_test.go

```diff
@@ -0,0 +1,251 @@
+	jsonData, reqJSON, err := RequestToJSON(req, "")
+	require.NoError(t, err)
+
+	// Check that the body is treated as a raw string
+	require.Equal(t, body, reqJSON.Body)
+	require.Contains(t, string(jsonData), `"body":"name=John+Doe\u0026age=30"`)
+	require.Contains(t, string(jsonData), `"Content-Type":["application/x-www-form-urlencoded"]`)
+}
+
+func TestRequestToJSON_MultipleHeadersWithSameKey(t *testing.T) {
+	// Create a GET request with repeated headers
+	req, err := http.NewRequest("GET", "http://example.com/path", nil)
+	require.NoError(t, err)
+	req.Header.Add("X-Forwarded-For", "192.168.1.1")
+	req.Header.Add("X-Forwarded-For", "10.0.0.1")
+
+	// Call the function
+	jsonData, reqJSON, err := RequestToJSON(req, "")
+	require.NoError(t, err)
+
+	// Check the headers in the struct
+	expectedHeaders := map[string][]string{
+		"X-Forwarded-For": {"192.168.1.1", "10.0.0.1"},
+	}
+	require.Equal(t, expectedHeaders, reqJSON.Headers)
+
+	// Check the JSON output
+	require.Contains(t, string(jsonData), `"X-Forwarded-For":["192.168.1.1","10.0.0.1"]`)
+}
+
+func TestRequestToJSON_InvalidJSONBody(t *testing.T) {
+	// Create a POST request with invalid/malformed JSON
+	body := `{"name": "John Doe", "age":`
+	req, err := http.NewRequest("POST", "http://example.com/path", strings.NewReader(body))
+	require.NoError(t, err)
+	req.Header.Add("Content-Type", "application/json")
+
+	// Call the function
+	jsonData, reqJSON, err := RequestToJSON(req, "")
+	require.NoError(t, err)
+
+	// Check that the raw body is still captured as a string
+	require.Equal(t, body, reqJSON.Body)
+
+	// Check the JSON output
+	require.Contains(t, string(jsonData), `"body":"{\"name\": \"John Doe\", \"age\":`)
+}
+
+func TestRequestToJSON_MultipartFormData(t *testing.T) {
+	// Create a buffer for the multipart form data
+	body := &bytes.Buffer{}
+	writer := io.MultiWriter(body)
+	_, err := writer.Write([]byte("----WebKitFormBoundary\nContent-Disposition: form-data; name=\"file\"; filename=\"test.txt\"\n\nfilecontent\n----WebKitFormBoundary--"))
+	require.NoError(t, err)
+
+	// Create a POST request with multipart/form-data
+	req, err := http.NewRequest("POST", "http://example.com/upload", body)
+	require.NoError(t, err)
+	req.Header.Add("Content-Type", "multipart/form-data; boundary=----WebKitFormBoundary")
+
+	// Call the function
+	jsonData, reqJSON, err := RequestToJSON(req, "")
+	require.NoError(t, err)
+
+	// Verify that the body is captured as a raw string
+	require.Contains(t, reqJSON.Body, "Content-Disposition: form-data; name=\"file\"; filename=\"test.txt\"")
+	require.Contains(t, reqJSON.Body, "filecontent")
+
+	// Check the JSON output
+	require.Contains(t, string(jsonData), `"Content-Type":["multipart/form-data; boundary=----WebKitFormBoundary"]`)
+}
+
+func TestRequestToJSON_ErrorHandling(t *testing.T) {
+	// Simulate a request with an unreadable body
+	body := &readErrorCloser{}
+	req, err := http.NewRequest("POST", "http://example.com/path", body)
```

Comment:
We don't need this `readErrorCloser` additional struct here. We can use `iotest.ErrReader` for it.
```suggestion
	req, err := http.NewRequest("POST", "http://example.com/path", iotest.ErrReader(errors.New("read error")))
```